xla/service/gpu/runtime/dynamic_slice_thunk.cc

```diff
@@ -180,7 +217,28 @@ absl::Status DynamicSliceThunk::ExecuteOnStream(const ExecuteParams& params) {
         VLOG(2) << "  - arg " << argument_idx << "[" << offset_idx
                 << "]: constant offset = " << *const_offset;
         offset_value(argument_idx, offset_idx) = *const_offset;
-
+      } else if (HloModule** offset_module = std::get_if<HloModule*>(&offset)) {
+        VLOG(0) << "Offset module: " << (*offset_module)->ToString();
+        TF_ASSIGN_OR_RETURN(auto replica_id,
+                            GetReplicaId(params.collective_params));
+
+        // Access indvar behind a lock.
+        indvar_mutex.Lock();
```

Comment:
How about using the same style in both places where you lock the mutex? E.g. using a nested block in both places.